core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1813,4 +1814,128 @@ class PlaintextConsumerTest extends BaseConsumerTest {
+    consumer3.assign(asList(tp))
+    consumer3.seek(tp, 1)
+
+    val numRecords1 = consumer1.poll(Duration.ofMillis(5000)).count()
+    val numRecords2 = consumer2.poll(Duration.ofMillis(5000)).count()
+    val numRecords3 = consumer3.poll(Duration.ofMillis(5000)).count()
+
+    consumer1.close()
+    consumer2.close()
+    consumer3.close()
+
+    assertTrue("Expected one consumer to consumer from earliest, one from latest, and one from offset 1 " +
+      "without fetching the last committed offset.", numRecords1 == 3 && numRecords2 == 0 && numRecords3 == 2)
+  }
+
+  @Test
+  def testConsumingWithEmptyGroupId(): Unit = {
+    val topic = "test_topic"
+    val partition = 0;
+    val tp = new TopicPartition(topic, partition)
+    createTopic(topic, 1, 1)
+
+    TestUtils.waitUntilTrue(() => {
+      this.zkClient.topicExists(topic)
+    }, "Failed to create topic")
+
+    val producer = createProducer()
+    producer.send(new ProducerRecord(topic, partition, "k1".getBytes, "v1".getBytes)).get()
+    producer.send(new ProducerRecord(topic, partition, "k2".getBytes, "v2".getBytes)).get()
+    producer.send(new ProducerRecord(topic, partition, "k3".getBytes, "v3".getBytes)).get()
+    producer.close()
+
+    // consumer 1 uses the empty group id
+    val consumer1Config = new Properties(consumerConfig)
+    consumer1Config.put(ConsumerConfig.GROUP_ID_CONFIG, "")
+    consumer1Config.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer1")
+    consumer1Config.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)
+    val consumer1 = createConsumer(configOverrides = consumer1Config)
+
+    // consumer 2 uses the empty group id and consumes from latest offset if there is no committed offset
+    val consumer2Config = new Properties(consumerConfig)
+    consumer2Config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest")
+    consumer2Config.put(ConsumerConfig.GROUP_ID_CONFIG, "")
+    consumer2Config.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer2")
+    consumer2Config.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)
+    val consumer2 = createConsumer(configOverrides = consumer2Config)
+
+    // consumer 3 uses the empty group id and consumes from earliest offset if there is no committed offset
+    val consumer3Config = new Properties(consumerConfig)
+    consumer2Config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    consumer3Config.put(ConsumerConfig.GROUP_ID_CONFIG, "")
+    consumer3Config.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer3")
+    consumer3Config.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1)
+    val consumer3 = createConsumer(configOverrides = consumer3Config)
+
+    consumer1.assign(asList(tp))
+    consumer2.assign(asList(tp))
+    consumer3.assign(asList(tp))
+
+    val records1 = consumer1.poll(Duration.ofMillis(5000))
+    consumer1.commitSync()
+
+    val records2 = consumer2.poll(Duration.ofMillis(5000))
+    consumer2.commitSync()
+
+    val records3 = consumer3.poll(Duration.ofMillis(5000))
```

Comment:
The third iteration here feels redundant. Doesn't it just read from the committed offset like `consumer2`?

Reply:
Yes, I put it to verify the `auto.offset.reset` doesn't affect the behavior. I'll remove it in the new commit.